### PR TITLE
 Rel-5_0 Selenium updated test - Agent part 1

### PR DESCRIPTION
**Pull request by s7design**
Hi @mgruner ,

This is first part of updated Selenium tests for Agent modules. I changed to verified method, updated comments and added backend checks, where it was needed.

Please let me know if you have any suggestion.

Regards,
Zoran

### CODE REVIEW

File: scripts/test/Selenium/Agent/AgentCustomerInformationCenter.t [hunk 1]
```diff
@@ -36,15 +37,17 @@ $Selenium->RunTest(
...
+
+        # navigate to AdminCustomerInformationCenter screen
+        $Selenium->VerifiedGet("${ScriptAlias}index.pl?Action=AgentCustomerInformationCenter");
 
         # input search parameters
         $Selenium->find_element( "#AgentCustomerInformationCenterSearchCustomerID", 'css' )
             ->send_keys($TestCustomerUserLogin);
-        sleep 1;
-        $Selenium->find_element("//*[text()='$TestCustomerUserLogin']")->click();
+        $Selenium->WaitFor( JavaScript => 'return typeof($) === "function" && $("li.ui-menu-item:visible").length' );
```
Comment (mgruner) [inline scripts/test/Selenium/Agent/AgentCustomerInformationCenter.t]: :+1:

File: scripts/test/Selenium/Agent/AgentDashboardCommon.t [hunk 2]
```diff
@@ -31,8 +32,11 @@ $Selenium->RunTest(
             Password => $TestUserLogin,
         );
 
+        # get script alias
         my $ScriptAlias = $Kernel::OM->Get('Kernel::Config')->Get('ScriptAlias');
-        $Selenium->get("${ScriptAlias}index.pl?Action=AgentDashboard");
+
+        # navigate to AgentDashboard screen
+        $Selenium->VerifiedGetq("${ScriptAlias}index.pl?Action=AgentDashboard");
```
Comment (mgruner) [inline scripts/test/Selenium/Agent/AgentDashboardCommon.t]: Typo?

File: scripts/test/Selenium/Agent/AgentTicketAttachment.t [hunk 3]
```diff
@@ -88,13 +91,17 @@ $Selenium->RunTest(
             ValidID        => 1,
             UserID         => $TestUserID,
         );
+        $Self->True(
+            $TestCustomerUserID,
+            "CustomerUserAdd - $TestCustomerUserID",
+        );
 
         # create test phone ticket with attachment
```
Comment (mgruner) [inline scripts/test/Selenium/Agent/AgentTicketAttachment.t]: I think this is one of many examples where we create the ticket via Selenium, even though we don't want to test ticket creation. Would it not be better to just create it in the backend and then use Selenium to test what we actually want to test? That would also save time.

**Comment by mgruner** [comment]
Hello Zoran, welcome back! I hope you had a good Christmas time. Please see my inline comments.

**Comment by s7design** [comment]
Hello @mgruner ,

Thank you for your nice words and comments. You are right, there was indeed left out typo in AgentDashboardCommon.t, it is corrected now. 

In AgentTicketAttachment.t i left intentionally creation of attachment through out frontend, since i don't think we have it done in selenium tests, but saying that it is more appropriate to have that kind of testing where it should be, in AgentTicketPhone or Email. Added backend ticket, article and attachment creation for AgentTicketAttachment.t.

Please let me know if you agree with these changes.

Regards,
Zoran

File: scripts/test/Selenium/Agent/AgentTicketAttachment.t [hunk 4]
```diff
@@ -68,65 +98,11 @@ $Selenium->RunTest(
             Password => $TestUserLogin,
         );
 
-        my $ScriptAlias = $Kernel::OM->Get('Kernel::Config')->Get('ScriptAlias');
```
Comment (mgruner) [inline scripts/test/Selenium/Agent/AgentTicketAttachment.t]: Wow, nice simplification!

File: scripts/test/Selenium/Agent/AgentCustomerInformationCenter.t [hunk 5]
```diff
@@ -36,15 +37,17 @@ $Selenium->RunTest(
             Groups => ['admin'],
         ) || die "Did not get test user";
 
+        # get script alias
         my $ScriptAlias = $Kernel::OM->Get('Kernel::Config')->Get('ScriptAlias');
-        $Selenium->get("${ScriptAlias}index.pl?Action=AgentCustomerInformationCenter");
-        sleep 1;
```
Comment (mgruner) [inline scripts/test/Selenium/Agent/AgentCustomerInformationCenter.t]: Here we still need to wait for an AJAX response. I'll add that.

**Comment by mgruner** [comment]
Merged, thank you very much!

File: scripts/test/Selenium/Agent/AgentTicketAttachment.t (hunk 4 above)
Comment (s7design) [inline scripts/test/Selenium/Agent/AgentTicketAttachment.t]: The pleasure is ours, if you can see our progress :)

Zoran
Regards

**Pull request closed without merging**